### PR TITLE
Switch to AWS SDK Rust

### PR DESCRIPTION
**Pull request by phyber**
This PR switches the project from Rusoto to AWS SDK Rust.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -999,26 +1057,26 @@ dependencies = [
...
 name = "ryu"
-version = "1.0.5"
+version = "1.0.9"
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "71d301d4193d031abdd79ff7e3dd721168a9572ef3fe51a1517aba235bd8f86e"
+checksum = "73b4b750c782965c211b42f022f59af1fbceabdd026623714f104152f1ec149f"
 
 [[package]]
 name = "s3audit"
 version = "1.0.0"
```
Comment (sftim) [inline Cargo.lock:1066]: I'm guessing we'll bump to 1.1 or 2.0 after this?

**Review by sftim — APPROVED** [review]
Looks good to me.

File: Cargo.lock (hunk 1 above)
Comment (phyber) [reply Cargo.lock:1066]: Probably 2.0, since it was a big internal change. It's nice to have a marker for that.

**Review by mbevc1 — APPROVED** [review]
Not really used Rust that much, but bump looks good :+1:

**Pull request merged**